### PR TITLE
React-Lesson_4.1

### PR DESCRIPTION
**Pull request by katemikh**
React-Lesson_4.1

### CODE REVIEW

File: src/components/TodoList/TodoList.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+              onRemoveTodo={onRemoveTodo}
+            />
+          );
+      })}
+    </ul>
+  );
+};
+
+TodoList.propTypes = {
+  todoList: PropTypes.array,
```
Comment (mattcarlotta) [inline src/components/TodoList/TodoList.js]: Stretch Goal:
What is `todoList` an array of? An array of strings, numbers, or an object (shape) that has structure!? When it comes to arrays and objects, the more specific you are in their structure, the better type checking you'll have during run-time!

<details>
<summary>Click here for answer</summary>

```js
TodoList.propTypes = {
  todoList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    }),
  ),
  onRemoveTodo: PropTypes.func.isRequired,
}
```

</details>

File: src/components/TodoListItem/TodoListItem.js [hunk 2]
```diff
@@ -14,6 +15,11 @@ function TodoListItem({ todo, onRemoveTodo }) {
         </button>
       </li>
     );
-}
+};
+
+TodoListItem.propTypes = {
+  todo: PropTypes.object,
```
Comment (mattcarlotta) [inline src/components/TodoListItem/TodoListItem.js]: Stretch Goal:
What is `todo` an object of (what's its shape)?

File: src/components/InputWithLabel/InputWithLabel.js [hunk 3]
```diff
@@ -26,4 +27,12 @@ const InputWithLabel = ({  id, name, todoTitle, onInputChange, children }) => {
   );
 };
 
+InputWithLabel.propTypes = {
```
Comment (mattcarlotta) [inline src/components/InputWithLabel/InputWithLabel.js:35]: Requires change:
Missing `children` prop type (it’s a `node`).

**Review by mattcarlotta — APPROVED** [review]
You nearly nailed this assignment, great work Kate! 👍

On a side note, I noticed all of your props are marked as optional. This isn’t necessarily true, some of them should be marked as `.isRequired`!

### You’re missing one requirement as mentioned below, please address that and then you can merge in the PR.

I added some optional stretch goals if you want to take your code to the next .

**Comment by katemikh** [comment]
added required changes to React-Lesson_4.1 as per advise

**Pull request merged**